Engine/Component/ComponentParticleSystem.cpp

```diff
@@ -334,5 +359,39 @@ Component* ComponentParticleSystem::Clone(bool original_prefab) const
 
 void ComponentParticleSystem::Copy(Component * component_to_copy) const
 {
-	
-}
\ No newline at end of file
+	*component_to_copy = *this;
+	*static_cast<ComponentParticleSystem*>(component_to_copy) = *this;
+}
+
+void ComponentParticleSystem::Emit(size_t count)
+{
+	if (count < MAX_PARTICLES)
+	{
+		particles_number = count;
+		for (size_t i = 0; i < particles_number; i++)
+		{
+			RespawnParticle(particles[i]);
+		}
+	}
+
+}
+
+void ComponentParticleSystem::Play()
```

Comment:
This function Replays, what I mean? it resets the values of the particles at 0 so they spawn from a new position, a play button is to start from where it was, so the values to 0 should be at stop and a pause button so it's just playing=false, so play button only sets playing to true.